keyboard/src/modules/Keyboard.rb

```diff
@@ -551,12 +552,13 @@ def SetKeyboard(keyboard)
         return false # Error
       end
 
-      @ckb_cmd = "/bin/loadkeys #{loadkeys_devices} #{keymap}"
+      # loadkeys is already escaped and it is multiple params inside, so cannot be escaped here
+      @ckb_cmd = "/usr/bin/loadkeys #{loadkeys_devices} #{keymap.shellescape}"
 
       # X11 command...
       # do not try to run this with remote X display
       if Ops.greater_than(Builtins.size(@Apply), 0) && x11_setup_needed
-        @xkb_cmd = Ops.add(Ops.add(XVersion.binPath, "/setxkbmap "), @Apply)
+        @xkb_cmd = "#{File.join(XVersion.binPath, "setxkbmap")} #{@Apply.shellescape}"
```

Comment:
Looks like those parameters need to be escaped separately, not in `@Apply` as a whole.